### ISSUE TITLE
First time user, no success with ICSM NTv2 Transformer within QGIS 3.2.3-Bonn

### ISSUE THREAD

**Issue by jhaasdykDFSI** [icsm-au/icsm_qgis_transformer#8]
I haven't used this transformer before, on QGIS 2.x or 3.x

My first attempts at using this on QGIS 3.2.3-Bonn are unsuccessful.
The transformer does download the GSB files and appears to transform them. 

However, QGIS Plugin ICSM NTv2 Transformer does NOT appear to return a dataset any different than GDA94. Using a GDA94 (EPSG 4283) dataset as input, the plugin reports successful transformation and outputs a file with GDA2020 .prj information, but the data itself is unchanged: It is in the same location as my GDA94 data, and a different location to other GDA2020 data that I employ. 
When I later defined a custom OTF transformation in QGIS between GDA94 <=> GDA2020, then the QGIS ICSM NTv2 Transformer product moves SW of the original GDA94 data (as it should, since it is defined as GDA2020, but to an incorrect location), while my own GDA2020 does as expected and becomes coincident with the underlying GDA94 dataset. 

Images provided showing coincident GDA94 and GDA2020 datasets, but offset QGIS dataset, after employing an OTF transformation

<img width="960" alt="qgis_icsm_transformer_displayed_with_otf_tf" src="https://user-images.githubusercontent.com/43433569/47836535-fafe1580-dda8-11e8-89a1-ea04bf2e17d3.PNG">
<img width="960" alt="qgis_icsm_transformer_displayed_without_otf_tf" src="https://user-images.githubusercontent.com/43433569/47836536-fafe1580-dda8-11e8-8c86-650ab1377275.PNG">

**Comment by jhaasdykDFSI** [comment]
The images in the earlier post have loaded out of order.
The Bottom image is WIHTOUT OTF Tf;
The Top image is WITH OTF Tf.

**Comment by jhaasdykDFSI** [comment]
I thought I would give the ICSM NTv2 Transformation one more try today.
This time, success. 
I can't reproduce my earlier failures.

As far as I know, all my settings are the same.
I haven't moved any files / folders around.
My previous attempts were all undertaken on a previous day.
I have since set up OTF transformations, but in a completely different area of QGIS (C:\OSGeo4W64\share\proj) and using a differently named (but same content) GSB file for the OTF transformation.

So, now it works for me, but I am a little cautious...

<img width="960" alt="qgis_icsm_transformer_displayed_with_otf_tf_latersuccessfulattempt" src="https://user-images.githubusercontent.com/43433569/47837307-d6f00380-ddab-11e8-807e-fefa2c61ad95.PNG">

**Comment by thillyard** [comment]
I can't help you with your issue, but i would love to hear how you got the OTF Transformation to work. I have been trying to get this to work using the transformation grids, but i can't get it to work. Can you elaborate on how you did this?

**Comment by alexgleith** [comment]
The OTF transformation should _not_ work.

As far at Proj understands, the two datums are the same... and practically, they are, it's just that the coordinates of data on the datums is different (correct me if I'm wrong here, someone!).

Oh, if you're doing OTF using the GSBs then it might work!

**Comment by thillyard** [comment]
You are correct Alex, the underlying parameter for both datums are exactly the same, the only difference now is that Australia has moved nearly 2metres since 1994...

Off topic for the transformation tool, but do you how we can include the icsm grid gsbs in Qgis for otf transformation. Or even the helmert parameters?

**Comment by alexgleith** [comment]
Hey @thillyard, I don't know how to include the GSB grid files for OTF, I haven't looked into it.

And regarding helmert parameters, it should be possible, but it won't be very good. The GSB files are needed because the 7 parameter transform isn't the same across Australia. You'd pretty much be better off not doing a transform, than a bad one!

**Comment by jhaasdykDFSI** [comment]
As noted by alexgleith and thillyard, unfortunately in EPSG
GDA94 =WGS = GDA2020.
Strictly speaking, this is correct, given that the accuracy of these transformations is listed as 3m. 
In practice, this is going to cause us a world of pain, as GDA94 != GDA2020 at any reasonable accuracy.

As for the 7 parameter GDA94 <=> GDA2020 Transformation being 'not-very good'...
Actually the 7 parameter is an accurate an concise method for defining the 
(a) REGULAR changes in the underlying ITRF92 => ITRF2014 improvements on which GDA is built
(b) REGULAR changes due to tectonic movement (~7cm per year, but varies across Australia).
Some states and territories in Australia have kept their GDA94 coordinates up to date and the only real change will be described well by 7 parameters. Some states need to account for residual distortion (we can make much more precise geodetic measurements now in this age of GNSS, than we could when GDA94 was defined). Some datasets (e.g. CORSnet-NSW would be suitable for a 7 parameter transformation UNLESS then have been 'localized' to local survey control.

Options are:
3D Transformation via 7-parametres (in Cartesian X,Y,Z)
2D Transformation via NTv2 grid in Lat, Long, "Conformal" GSB file = 7 parameter Tf!!
2D Transformation via NTv2 grid in Lat, Long, "Conformal plus Distortions" GSB file.

**Comment by jhaasdykDFSI** [comment]
@alexgleith and @thillyard

I found this solution for setting up a new OTF transformation in QGIS.
This didn't require any direct changes to PROJ, but rather just modified the srs.db called by QGIS and pointed to the GSB files.
https://gis.stackexchange.com/questions/298011/how-to-use-gridfiles-gsb-in-qgis/298027

From the looks of the srs.db, it would be similarly easy to add/modify a 7-parameter transformation.

**Comment by thillyard** [comment]
@jhaasdykDFSI Thanks so much for this, i have successfully incorporated this as an OTF transformation in QGIS. It is successful for GDA94 to GDA2020 (project datum GDA94), but there still is an issue for a project with a datum of GDA2020. Still looking into why.

**Comment by jhaasdykDFSI** [comment]
No worries.
I needed to define the custom transformation for EACH projection I might be using.
Defining GDA94 <=> GDA2020 is not, apparently, sufficient (as it is in ArcGIS for example).


GDA94 <=> GDA2020 MGA Zone 55
GDA94 <=> GDA2020 MGA Zone 56
GDA2020 <=> GDA94 MGA Zone 55
GDA2020 <=> GDA94 MGA Zone 56
etc.
are also required depending on your Input data and Project Datum.
I'll find the exact EPSG examples when I get time.

I also added WGS84 <=> GDA2020... as a bit of a 'unofficial' transformation method to force my WMS onto my GDA2020 Project Datum.
